_studio/mfx_lib/shared/src/mfx_mpeg2_encode_vaapi.cpp

```diff
@@ -380,18 +387,74 @@ mfxStatus VAAPIEncoder::QueryEncodeCaps(ENCODE_CAPS & caps)
     caps.VCMBitrateControl = 0; // Video conference mode
     caps.HeaderInsertion   = 0; // We will provide headers (SPS, PPS) in binary format to the driver
     caps.MbQpDataSupport   = 1;
-    caps.SkipFrame         = 1;
 
     caps.SliceIPBOnly      = 1;
-    caps.MaxNum_Reference  = 1;
-    caps.MaxPicWidth       = 1920;
-    caps.MaxPicHeight      = 1088;
     caps.NoInterlacedField = 0; // Enable interlaced encoding
-    caps.SliceStructure    = 1; // 1 - SliceDividerSnb; 2 - SliceDividerHsw; 3 - SliceDividerBluRay; the other - SliceDividerOneSlice
 
+    std::map<VAConfigAttribType, size_t> idx_map;
+    VAConfigAttribType attr_types[] = {
+        VAConfigAttribMaxPictureHeight,
+        VAConfigAttribMaxPictureWidth,
+        VAConfigAttribEncSkipFrame,
+        VAConfigAttribEncMaxRefFrames,
+        VAConfigAttribEncSliceStructure
+    };
+
+    std::vector<VAConfigAttrib> attrs;
+    attrs.reserve(sizeof(attr_types) / sizeof(attr_types[0]));
+
+    for (size_t i=0; i < sizeof(attr_types)/sizeof(attr_types[0]); ++i)
+    {
+        attrs.push_back( createVAConfigAttr(attr_types[i], 0) );
```

Comment:
stupid question: why direct `{attr_types[i], 0}` bypassing `createVAConfigAttr` won't work? why do you need `createVAConfigAttr`?

Reply:
I also don't like this function) I believe it won't work without proper casting. If you cast ```{attr_types[i], 0}``` to ```std::initializer_list<VAConfigAttrib>``` it should work:

```
attrs.emplace_back(std::initializer_list<VAConfigAttrib>(attr_types[i], 0));
```